Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -17,296 +17,516 @@
 #import <StubReturn.h>
 #import <Starboard.h>
 #import <math.h>
+
+#import "CGSurfaceInfoInternal.h" // TODO(DH) Evaluate the need for this header.
+
 #import <CoreGraphics/CGContext.h>
 #import <CoreGraphics/CGPath.h>
 #import <CoreGraphics/CGLayer.h>
-#import "CGColorSpaceInternal.h"
-#import "CGContextInternal.h"
-#include "LoggingNative.h"
-#import "_CGLifetimeBridgingType.h"
 #import <CoreGraphics/CGAffineTransform.h>
 #import <CoreGraphics/CGGradient.h>
-#import <Foundation/NSString.h>
-#import <UIKit/UIImage.h>
-#import <UIKit/UIFont.h>
-#import <UIKit/UIColor.h>
-#import "CGSurfaceInfoInternal.h"
-#import <pthread.h>
+#import "CGColorSpaceInternal.h"
+#import "CGContextInternal.h"
 
-static const wchar_t* TAG = L"CGContext";
+#import <CFRuntime.h>
+
+#include <COMIncludes.h>
+#import <d2d1.h>
+#import <d2d1_1.h>
+#import <d2d1effects_2.h>
+#import <wrl/client.h>
+#include <COMIncludes_end.h>
+#import <LoggingNative.h>
+
+#import <list>
+#import <vector>
+#import <stack>
+
+using namespace Microsoft::WRL;
+
+static inline D2D_RECT_F __CGRectToD2D(CGRect rect) {
+    return {
+        rect.origin.x, rect.origin.y, rect.origin.x + rect.size.width, rect.origin.y + rect.size.height,
+    };
+}
+
+struct __CGContextDrawingState {
+    // This is populated when the state is saved, and contains the D2D parameters that CG does not know.
+    ComPtr<ID2D1DrawingStateBlock> d2dState{ nullptr };
+
+    // Fills
+    ComPtr<ID2D1Brush> fillBrush{ nullptr };
```

Comment:
I know :/; I'd like to specify explicit defaults here even if they can be inferred by default construction.

---
In reply to: [83091463](https://github.com/Microsoft/WinObjC/pull/1126#discussion_r83091463) [](ancestors = 83091463)